src/javascript/app/pages/user/metatrader/metatrader.config.js

```diff
@@ -119,10 +119,21 @@ const MetaTraderConfig = (() => {
             const $new_account_financial_authenticate_msg = $('#new_account_financial_authenticate_msg');
             $new_account_financial_authenticate_msg.setVisibility(0);
             const is_virtual = Client.get('is_virtual');
+            const is_demo = accounts_info[acc_type].is_demo;
 
             if (!Client.get('currency')) {
                 resolve($messages.find('#msg_set_currency').html());
-            } else if (is_virtual && !accounts_info[acc_type].is_demo) { // virtual clients can only open demo MT accounts
+            } else if (is_demo) {
+                BinarySocket.wait('get_settings').then(() => {
+                    const response_get_settings = State.getResponse('get_settings');
+                    if (response_get_settings.residence === 'United Kingdom' && response_get_settings.date_of_birth === null) {
+                        $messages.find('#msg_age_verification').setVisibility(1);
+                        resolve($messages.find('#msg_age_verification').html());
+                    }
+                });
+
+                resolve($messages.find('#msg_age_verification').html());
```

Comment:
@ahkuma won't this resolve without waiting for get_settings? how will the get settings resolve get called then? doesn't a promise only resolve once?